src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2KeepAlive.cs

```diff
@@ -0,0 +1,99 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Net.Http.Headers;
+using System.Net.Http.HPack;
+using System.Net.Security;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+
+namespace System.Net.Http
+{
+    internal sealed partial class Http2Connection
+    {
+        internal enum KeepAliveState
+        {
+            None,
+            PingSent
+        }
+
+        internal class Http2KeepAlive
```

Comment:
I would not put this in a separate class; it'd be better to bake it all into `Http2Connection`.

Reply:
In ASP.NET Core I put it in a separate class to reduce server connection memory usage when keep alive is not enabled. I don't know if that is a consideration here.

Reply:
Since this is per-connection and each connection handles 100+ requests, I'm not _super_ concerned by resource usage of a handful of fields.

Reply:
the extra class is to separate the logic which is related only to keep alive and not for the rest of connection.

Reply:
the class was removed